src/ui/layer_bar.ts

```diff
@@ -168,6 +190,21 @@ class LayerWidget extends RefCounted {
     }
     title += ", drag to move, shift+drag to copy";
     element.title = title;
+    // Color widget updates
+    if (panel.layerGroupViewer.viewerState.enableLayerColorWidget.value) {
+      if (layer.supportsLayerBarColorSyncOption) {
+        const color = this.layer.layerBarColor;
+        if (color) {
+          element.dataset.color = "fixed";
+          layerColorElement.style.backgroundColor = color;
+        } else {
+          element.dataset.color = "rainbow";
+        }
+      } else {
+        element.dataset.color = "unsupported";
```

Comment:
If the layer type switches from supported to unsupported, the old color gets left around. I suggest we add something like

```
        // Reset color to default - otherwise the grid pattern for
        // unsupported layers will be overlayed on top of the old color.
        // this can happen if the layer changes type from a supported to
        // an unsupported type.
        layerColorElement.style.backgroundColor = "";
```

Reply:
This might also need to happen in `src/ui/layer_list_panel.ts`, I didn't check that due to layer type changes not updating the styling